notebooks/advanced/Time_Series_Forecasting_with_Unstructured_Data_and_Amazon_SageMaker_Neural_Topic_Model/2_NTM.ipynb

```diff
@@ -0,0 +1,838 @@
+    "\n",
+    "from sagemaker.predictor import RealTimePredictor, csv_serializer, json_deserializer\n",
+    "\n",
+    "\n",
+    "endpoint_widget = util.create_text_widget( \"endpoint_name\", \"ONLY enter if you have already (previously) deployed the NTM model. It should start with 'ntm-\" )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "endpoint = endpoint_widget.value\n",
+    "assert endpoint, \"Endpoint not set.\"\n",
+    "\n",
+    "ntm_predictor = RealTimePredictor(endpoint, sagemaker_session=session, serializer=csv_serializer, deserializer=json_deserializer, content_type='text/csv')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### New Deployment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#execute if you have NOT deployed the NTM model for this notebook series and you skipped the previous two cells\n",
+    "#(didn't enter anything into the widget box)\n",
+    "from sagemaker.predictor import csv_serializer, json_deserializer\n",
+    "\n",
+    "\n",
+    "ntm_predictor = ntm.deploy(initial_instance_count=1, instance_type='ml.m4.xlarge')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Test the Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ntm_predictor.content_type = 'text/csv'\n",
+    "ntm_predictor.serializer = csv_serializer\n",
+    "ntm_predictor.deserializer = json_deserializer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "To do a \"sanity\" check that our topic model is working as expected, we look at the extracted topic vectors from the titles and check if the topic distribution of the title is similar to that of the corresponding headline. Remember that our model has not seen the titles before. As a measure of \"similarity\", we compute the cosine similarity for a random title and associated headline. A high cosine similarity indicates that Titles and Headlines have a similar representation in this low dimensional embedding space. \n",
+    "\n",
+    "A cosine similarity of the Title-Headline can also be used as a feature: well written titles that correlate well with the actual Headline may obtain a higher popularity score. This could be used to check if titles and headlines represent the content of the document accurately, but we will not explore this further in this notebook.\n",
+    "\n",
+    "We also visualize the Headlines in a T-SNE plot to capture the number of distinct Topic clusters that appear."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%time\n",
```

Comment:
If this magic command is intended to measure the time required to execute this cell, should use "%%time" instead.